workers/src/bin/dvn.rs

```diff
@@ -0,0 +1,118 @@
+//! Main offchain workflow for Nuff's DVN.
+
+use alloy::primitives::{Address, U256};
+use eyre::Result;
+use futures::stream::StreamExt;
+use tracing::{debug, error, info};
+use tracing_subscriber::EnvFilter;
+use workers::{
+    abi::{L0V2EndpointAbi::PacketSent, SendLibraryAbi::DVNFeePaid},
+    chain::{
+        connections::{build_subscriptions, get_abi_from_path, get_http_provider},
+        contracts::{create_contract_instance, query_already_verified, query_confirmations, verify},
+    },
+    data::dvn::Dvn,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Initialize tracing
+    tracing_subscriber::fmt()
+        .without_time()
+        .with_target(false)
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    // Initialize the DVN worker.
+    let mut dvn_data = Dvn::new_from_env()?;
+
+    // Create the WS subscriptions for listening to the events.
+    let (_provider, mut endpoint_stream, mut sendlib_stream) = build_subscriptions(&dvn_data.config).await?;
+
+    // Create an HTTP provider to call contract functions.
+    let http_provider = get_http_provider(&dvn_data.config)?;
+
+    // Get the relevant contract ABI, and create contract.
+    let receivelib_abi = get_abi_from_path("./abi/ArbitrumReceiveLibUln302.json")?;
+    let contract_address = dvn_data.config.receivelib_uln302_addr.parse::<Address>()?;
+    let receivelib_contract = create_contract_instance(contract_address, http_provider, receivelib_abi)?;
+
+    info!("Listening to chain events...");
+
+    loop {
+        dvn_data.listening();
+        tokio::select! {
+            Some(log) = endpoint_stream.next() => {
+                match log.log_decode::<PacketSent>() {
+                    Err(e) => {
+                        error!("Received a `PacketSent` event but failed to decode it: {:?}", e);
+                    }
+                    Ok(inner_log) => {
+                        debug!("PacketSent event found and decoded");
+                        dvn_data.packet_received(inner_log.data().clone());
+                    },
+                }
+            }
+            Some(log) = sendlib_stream.next() => {
+                match log.log_decode::<DVNFeePaid>() {
+                    Err(e) => {
+                        error!("Received a `DVNFeePaid` event but failed to decode it: {:?}", e);
+                    }
+                    Ok(inner_log) => {
+                        if dvn_data.packet.is_some() {
```

Comment:
```suggestion
                    Ok(inner_log)  if dvn_data.packet.is_some() {
```
Needs the other bracket removed below